### DISCUSSION TITLE
Right-click options to delete/move/edit devices when making changes to Racks

### DISCUSSION THREAD

**Discussion by dsbounxou** [netbox-community/netbox#17864] [Q&A] [answered]
We are using ver4.0.5, is there an easier way to edit/move/delete devices in the Racks? Currently if I wanted to delete several devices, I am clicking on each one which takes me into the device. I then delete it, but when I do that it kicks me back out to the devices list. I then have to click through to get back to my rack I am working on and repeat the process.

**Comment by AADIP** [comment]
From the rack view go to "Devices" that is under related objects below the rack's elevation view. This puts you in the devices table again, but with the rack filter. While in this view, multi-select devices and at the bottom of table is "Delete Selected".

**Comment by candlerb** [comment, marked as answer]
Suggestion: From the rack view, right click on device -> "Open in new tab". Switch to that tab, delete the device, close the tab.

It's not pretty, but it's functional. (I use this approach if I have a bunch of IP addresses I need to change; just open them up in lots of tabs, and then change them)

**Comment by zmaster7** [comment]
For bulk deletions, you can look under the "related objects" section in the rack view and open said devices in a new tab. This will give you a list of devices filtered by the rack, which you can then multi-select and bulk delete (after verifying). Depending on what you're trying to edit, you can bulk edit here too.

If you are comfortable using plugins, there is a certified plugin for reordering rack devices, though I have not tried it out myself yet: [https://github.com/netbox-community/netbox-reorder-rack](https://github.com/netbox-community/netbox-reorder-rack)
